src/guacd/connection.c

```diff
@@ -278,10 +278,18 @@ static int guacd_route_connection(guacd_proc_map* map, guac_socket* socket) {
         proc = guacd_proc_map_retrieve(map, identifier);
         new_process = 0;
 
-        /* Warn if requested connection does not exist */
-        if (proc == NULL)
-            guacd_log(GUAC_LOG_INFO, "Connection \"%s\" does not exist.",
-                    identifier);
+        /* Warn and ward off client if requested connection does not exist */
```

Comment:
What about the case where a particular protocol is requested (for a new connection) and that protocol is not defined?

Reply:
This isn't handled here - this code path is specifically for existing connections (i.e. selecting $<connection_id>)